### PR TITLE
feat: support command 'kpm package' to package a kcl package into tar.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #20

#### 2. What is the scope of this PR (e.g. component or file name):

kpm

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

Support command 'kpm package' to package a kcl package into tar.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by Peefy** [comment]
Pls rebase the main branch.

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+var ignores = []string{".git", ".tar"}
+
+func TarDir(srcDir string, tarPath string) error {
```
Comment (Peefy) [inline pkg/utils/utils.go:112]: Can this be a standard OC tarball?

File: pkg/opt/opt.go [hunk 2]
```diff
@@ -7,3 +7,19 @@ type InitOptions struct {
 	Name     string
 	InitPath string
 }
+
+type AddOptions struct {
```
Comment (Peefy) [inline pkg/opt/opt.go:11]: More documents

File: pkg/cmd/cmd_add.go [hunk 3]
```diff
@@ -91,5 +90,10 @@ func onlyOnceOption(c *cli.Context, name string) (*string, error) {
 }
 
 func addGitDep(opt *opt.AddOptions, kclPkg *pkg.KclPkg) error {
-	return ops.KpmAdd(opt, kclPkg)
+	if opt.RegistryOpts.Git == nil {
```
Comment (Peefy) [inline pkg/cmd/cmd_add.go:93]: There is usually a unified method for checking these parameters. For example:
https://github.com/KusionStack/kusion/blob/main/pkg/cmd/init/options.go#L53

File: pkg/package/package.go [hunk 4]
```diff
@@ -157,7 +158,7 @@ func getDeps(deps modfile.Dependencies, lockDeps modfile.Dependencies, localPath
 		// download dependencies
 		lockedDep, err := d.Download(dir)
 		if err != nil {
-			return nil, fmt.Errorf("checksum mismatch")
+			return nil, fmt.Errorf("kpm: failed download dependency")
```
Comment (Peefy) [inline pkg/package/package.go:161]: var FailedDownloadError = errors.New("kpm: failed download dependency")

File: pkg/cmd/cmd_pkg.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 The KCL Authors. All rights reserved.
```
Comment (Peefy) [inline pkg/cmd/cmd_pkg.go:1]: s/Copyright 2021/Copyright 2023

File: pkg/cmd/cmd_pkg.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"os"
+
+	"github.com/urfave/cli/v2"
+	pkg "kusionstack.io/kpm/pkg/package"
+	"kusionstack.io/kpm/pkg/reporter"
+)
+
+// NewAddCmd new a Command for `kpm add`.
```
Comment (Peefy) [inline pkg/cmd/cmd_pkg.go:13]: s/NewAddCmd/NewPkgCmd

Comment (Peefy) [inline pkg/cmd/cmd_pkg.go:13]: Wrong docs.

File: pkg/cmd/cmd_pkg.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+func NewPkgCmd() *cli.Command {
+	return &cli.Command{
+		Hidden: false,
+		Name:   "package",
```
Comment (Peefy) [inline pkg/cmd/cmd_pkg.go:17]: s/package/pkg

**Review by Peefy — COMMENTED** [review]
Use `kpm pkg` instead of `kpm package`

File: pkg/utils/utils.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/utils/utils.go:113]: fixed.

File: pkg/cmd/cmd_add.go (hunk 3 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_add.go:100]: fixed.

File: pkg/cmd/cmd_pkg.go (hunk 5 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_pkg.go]: fixed.

File: pkg/cmd/cmd_pkg.go (hunk 6 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_pkg.go]: fixed.

Comment (zong-zhe) [reply pkg/cmd/cmd_pkg.go]: fixed.

File: pkg/cmd/cmd_pkg.go (hunk 7 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_pkg.go]: fixed.

File: pkg/package/package.go (hunk 4 above)
Comment (zong-zhe) [reply pkg/package/package.go]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**